src/libNetwork/P2PComm.cpp

```diff
@@ -373,9 +376,25 @@ void SendJobPeers<T>::DoSend() {
       continue;
     }
 
-    SendMessageCore(peer, m_message, m_startbyte, m_hash);
+    // Launch the thread.
+    auto hThread = make_shared<std::future<void>>(
+        std::async(std::launch::async, SendMessageCore, peer, m_message,
+                   m_startbyte, m_hash));
+
+    // Park the thread if message sending taking more than expected
+    auto status = hThread->wait_for(std::chrono::seconds(5));
+
+    if (status == std::future_status::timeout) {
+      LOG_GENERAL(WARNING, "Sending message to "
```

Comment:
In the interest of brevity, I suggest to shorten to: "Sending delayed for <IP>"
Anyway the presence of the `LOG_MARKER` will already help us find the scope for this entire execution in the logs.

Reply:
sure. but i still prefer the message logging for peer, ofcourse short one like you suggested.